target/ppc/excp_helper.c

```diff
@@ -1709,13 +1709,13 @@ static void ppc_hw_interrupt(CPUPPCState *env)
      * clear when coming out of some power management states (in order
      * for them to become a 0x100).
      */
-    async_deliver = (msr_ee != 0) || env->resume_as_sreset;
+    async_deliver = !!(env->msr & M_MSR_EE) || env->resume_as_sreset;
```

Comment:
`async_deliver` is a boolean and the msr result is doing a logical or, so I don't think the `!!` is needed